### PR TITLE
creating short version of PR#204

### PR DESCRIPTION
**Pull request by cippy**
As the title says, this is a subset of PR#204, with most of its optimisations, but without the changes affecting the fakes for now.

Hopefully this can be merged immediately, so that then reviewing PR#204 becomes simpler.
I saw no issue running locally, and CI was working fine also for PR#204.

### CODE REVIEW

File: utilities/boostHistHelpers.py [hunk 1]
```diff
@@ -109,6 +109,13 @@ def addHists(h1, h2, allowBroadcast=True):
 def sumHists(hists):
     return reduce(addHists, hists)
 
+def addHistsNoCopy(h1, h2, allowBroadcast=True):
```
[the remark this answers is not in the document]
Comment (cippy) [reply utilities/boostHistHelpers.py:112]: ok, having an additional "if" will not slow things down

File: wremnants/CardTool.py [hunk 2]
```diff
@@ -740,7 +740,13 @@ def writeHist(self, h, name, setZeroStatUnc=False, decorrByBin={}, hnomi=None):
             axes = self.project[:]
             if "charge" in h.axes.name and not self.xnorm:
                 axes.append("charge")
-            h = h.project(*axes)
+            # don't project h into itself when axes to project are all axes
+            if any (ax not in h.axes.name for ax in axes):
```
[the remark this answers is not in the document]
Comment (davidwalter2) [reply wremnants/CardTool.py:744]: I think it should be ```if not all(ax in h.axes.name for ax in axes)``` 
Or ```if not set(axes).issubset(h.axes.name)```

**Review by davidwalter2 — COMMENTED** [review]
For me the changes look good, I only made one comment on top of the one from Kenneth but it's only a detail. 

Before we merge, could you please pull the master to see if the impacts are exactly the same? There has been an update yesterday

**Comment by cippy** [comment]
I modified the default of createNew from False to True, since this is how it was used previously. Using False as default is dangerous because if the input is not a copy of some object read from somewhere, the original object is modified and this might lead to undesired behaviour.
Instead, it is better to call the functions with createNew=False explicitly where needed, so that one is aware of what's happening

**Pull request merged**